Clears application/run/main_scene from project.godot if the file is deleted in editor.

twinderw opened the pull request:
Fixes a bug relating to the main scene. 

Currently, if the main scene is deleted through the editor, nothing is changed in the project.godot file to indicate this. Therefore the next time the project is loaded, it will try to open a file that has been deleted.

This change fixes the problem by checking if the file to be deleted is the main scene, and if so clearing "application/run/main_scene" from the project.godot file.

Fixes issue #22473

twinderw committed:
Clears application/run/main_scene if that file is deleted

Fixes issue #22473

twinderw committed:
Revert "Clears application/run/main_scene if that file is deleted"

This reverts commit 8e77ebfcc04ae1cf02d9747ebf3b6536e2ede2cd.

twinderw committed:
Revert "Revert "Clears application/run/main_scene if that file is deleted""

This reverts commit f53d56ed856496eb62d8b5fbe7e629355fd1f459.

LikeLakers2 commented on editor/dependency_editor.cpp:
```diff
@@ -495,6 +495,10 @@ void DependencyRemoveDialog::ok_pressed() {
+      // If the file we are deleting is the main scene, clear the info for the main scene in project.godot.
+      if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/run/main_scene")) {
+         ProjectSettings::get_singleton()->set("application/run/main_scene", "");
+      }
```
You should be using tabs to align, not spaces.

twinderw replied:
I forgot that I had my IDE set to use spaces. I'm re-submitting now.

twinderw committed:
Clears application/run/main_scene from project.godot if the file is deleted in editor.

Fixes a bug relating to the main scene.

Currently, if the main scene is deleted through the editor, nothing is changed in the project.godot file to indicate this. Therefore the next time the project is loaded, it will try to open a file that has been deleted.

This change fixes the problem by checking if the file to be deleted is the main scene, and if so clearing "application/run/main_scene" from the project.godot file.

Fixes issue #22473
